seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/ConfigBuilder.java

```diff
@@ -130,49 +129,37 @@ private boolean checkIsContainHive() {
     /**
      * create plugin class instance, ignore case.
      **/
-    private <T extends Plugin<ENVIRONMENT>> T createPluginInstanceIgnoreCase(String name, PluginType pluginType) throws Exception {
-        if (name.split("\\.").length != 1) {
-            // canonical class name
-            return (T) Class.forName(name).newInstance();
+    @SuppressWarnings("unchecked")
+    private <T extends Plugin<ENVIRONMENT>> T createPluginInstanceIgnoreCase(String pluginName, PluginType pluginType) throws Exception {
+        Map<PluginType, Class<?>> pluginBaseClassMap = engine.getPluginTypes();
+        if (!pluginBaseClassMap.containsKey(pluginType)) {
+            throw new IllegalArgumentException("PluginType not support : [" + pluginType + "]");
         }
-        String packageName;
-        ServiceLoader<T> plugins;
-        switch (pluginType) {
-            case SOURCE:
-                packageName = configPackage.getSourcePackage();
-                Class<T> baseSource = (Class<T>) Class.forName(configPackage.getBaseSourceClass());
-                plugins = ServiceLoader.load(baseSource);
-                break;
-            case TRANSFORM:
-                packageName = configPackage.getTransformPackage();
-                Class<T> baseTransform = (Class<T>) Class.forName(configPackage.getBaseTransformClass());
-                plugins = ServiceLoader.load(baseTransform);
-                break;
-            case SINK:
-                packageName = configPackage.getSinkPackage();
-                Class<T> baseSink = (Class<T>) Class.forName(configPackage.getBaseSinkClass());
-                plugins = ServiceLoader.load(baseSink);
-                break;
-            default:
-                throw new IllegalArgumentException("PluginType not support : [" + pluginType + "]");
+        Class<T> pluginBaseClass = (Class<T>) pluginBaseClassMap.get(pluginType);
+        if (pluginName.split("\\.").length != 1) {
+            // canonical class name
+            Class<T> pluginClass = (Class<T>) Class.forName(pluginName);
+            if (pluginClass.isAssignableFrom(pluginBaseClass)) {
+                throw new IllegalArgumentException("plugin: " + pluginName + " is not extends from " + pluginBaseClass);
+            }
+            return pluginClass.newInstance();
```

Comment:
Note that 'newInstance()' is deprecated

Reply:
Done, use constructor to create instance.